### ISSUE TITLE
google-beta v3.87 crashes with monitoring_config on google_container_cluster

### ISSUE THREAD

**Issue by hazcod** [hashicorp/terraform-provider-google#10280]
When doing an `apply`, terraform crashes.

### Terraform Version

terraform v1.0.8
google-beta v3.87.0

### Affected Resource(s)
* google_container_cluster

### Terraform Configuration Files

```tf
resource "google_container_cluster" "main_cluster" {

  project  = var.project_id
  name     = var.cluster_name
  location = var.cluster_location

  # use stackdriver GKE for system and workload logs
  logging_service = "logging.googleapis.com/kubernetes"
  logging_config {
    enable_components = ["SYSTEM_COMPONENTS", "WORKLOADS"]
  }

  # use stackdriver GKE for system monitoring
  monitoring_service = "monitoring.googleapis.com/kubernetes"
  monitoring_config {
    enable_components = ["SYSTEM_COMPONENTS"]
  }

}
```

Logs:

```
Stack trace from the terraform-provider-google-beta_v3.87.0_x5 plugin:

panic: interface conversion: interface {} is []interface {}, not []string

goroutine 57 [running]:
github.com/hashicorp/terraform-provider-google-beta/google-beta.expandMonitoringConfig(...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-google-beta/google-beta/resource_container_cluster.go:3554
github.com/hashicorp/terraform-provider-google-beta/google-beta.resourceContainerClusterCreate(0x14001363180, 0x103b4f800, 0x140006f3800, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-google-beta/google-beta/resource_container_cluster.go:1548 +0x3cdc
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x14000532c40, 0x103bfd0e0, 0x14000376100, 0x14001363180, 0x103b4f800, 0x140006f3800, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.5.0/helper/schema/resource.go:318 +0x170
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x14000532c40, 0x103bfd0e0, 0x14000376100, 0x14001010150, 0x1400079bde0, 0x103b4f800, 0x140006f3800, 0x0, 0x0, 0x0, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.5.0/helper/schema/resource.go:456 +0x4ec
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x140007beff0, 0x103bfd0e0, 0x14000376100, 0x14000dfc0a0, 0x14000376100, 0x103a96080, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.5.0/helper/schema/grpc_provider.go:955 +0x6f8
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0x14000be9b40, 0x103bfd188, 0x14000376100, 0x14001010000, 0x14000be9b40, 0x1012b02d0, 0x1038e86a0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.2.1/tfprotov5/server/server.go:332 +0x94
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x103a96080, 0x14000be9b40, 0x103bfd188, 0x14000e0e570, 0x140003002a0, 0x0, 0x103bfd188, 0x14000e0e570, 0x1400100b500, 0x148e)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.2.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x1c8
google.golang.org/grpc.(*Server).processUnaryRPC(0x140004a96c0, 0x103c3eb98, 0x140001ab080, 0x14001008000, 0x140000a6a50, 0x104b99b00, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.40.0/server.go:1297 +0x3e8
google.golang.org/grpc.(*Server).handleStream(0x140004a96c0, 0x103c3eb98, 0x140001ab080, 0x14001008000, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.40.0/server.go:1626 +0xa50
google.golang.org/grpc.(*Server).serveStreams.func1.2(0x1400018e6b0, 0x140004a96c0, 0x103c3eb98, 0x140001ab080, 0x14001008000)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.40.0/server.go:941 +0x94
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.40.0/server.go:939 +0x1f8

Error: The terraform-provider-google-beta_v3.87.0_x5 plugin crashed!
```

**Comment by hazcod** [comment]
Seems to be fixed in v3.89.0! However...

```
googleapi: Error 400: Cannot specify logging_config or monitoring_config together with logging_service or monitoring_service., badRequest
```

```hcl
resource "google_container_cluster" "main_cluster" {

  # use stackdriver GKE for system and workload logs
  logging_service = "logging.googleapis.com/kubernetes"
  logging_config {
    enable_components = ["SYSTEM_COMPONENTS", "WORKLOADS"]
  }

  # use stackdriver GKE for system monitoring
  monitoring_service = "monitoring.googleapis.com/kubernetes"
  monitoring_config {
    enable_components = ["SYSTEM_COMPONENTS"]
  }

...
}
```

**Comment by rileykarson** [comment]
That issue comes from the API. We don't seem to run into it in our tests where we've set logging_config and logging_service together- I'm not sure why.

You may be able to set `logging_service` and `monitoring_service` to `""`, potentially?
